CMakeLists.txt

```diff
@@ -161,7 +160,8 @@ add_subdirectory(build_tools/embed_data/)
 
 add_subdirectory(iree/base)
 add_subdirectory(iree/hal)
-add_subdirectory(iree/modules)
+# Hal module is special and must be built before the compiler.
```

Comment:
I'd like to better understand the `add_subdirectory` CMake changes in this PR before merging.

Can you try reverting the `add_subdirectory` changes in the root and post what errors you see?

Reply:
Depending on things are layed out, either no error is thrown (we just fail to find statically linked dialects / passes), or we get the error for an undefined main function.

AFAICT its failing to statically link in required dependencies.